### PR TITLE
docs: Add Apache APISIX loves Rust

### PR DESCRIPTION
**Pull request by nfrankel**
Screenshot 2022-09-25 at 10 15 30

### CODE REVIEW

File: blog/en/blog/2022/09/28/rust-loves-apisix.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+However, if Lua is not part of your tech stack, [diving into a new ecosystem is a considerable investment](https://blog.frankel.ch/on-learning-new-programming-language/). Therefore, Apache APISIX offers developers to write plugins in several other languages. In this post, I'd like to highlight how to write such a plugin with Rust.
+
+## A bit of context
+
+Before I dive into the "how", let me first describe a bit of context surrounding the Rust integration in Apache APISIX. I believe it's a good story because it highlights the power of Open Source.
+
+It starts with the Envoy proxy.
+
+>Envoy is an open source edge and service proxy, designed for cloud-native applications
```
Comment (yzeng25) [inline blog/en/blog/2022/09/28/rust-loves-apisix.md]: I think we need a space here to show properly.
```suggestion
> Envoy is an open source edge and service proxy, designed for cloud-native applications
```

File: blog/en/blog/2022/09/28/rust-loves-apisix.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+* Developers can create SDK for their tech stack
+* Proxy and API Gateway providers can integrate `proxy-wasm` in their product
+
+## Apache APISIX and proxy-wasm
+
+The Apache APISIX project decided to integrate `proxy-wasm` into the product to benefit from the standardization effort. It also allows end-users to start with Envoy, or any other `proxy-wasm`-compatible reverse proxy, to migrate to Apache APISIX when necessary.
+
+APISIX doesn't implement `proxy-wasm` but integrates [wasm-nginx-module](https://github.com/api7/wasm-nginx-module). It's an Apache v2-licensed project provided by [api7.ai](https://api7.ai/), one of the main contributors to Apache APISIX. As its name implies, integration is done at the NGINX level.
+
+![Apache APISIX and WebAssemby architecture overview](https://api7.ai/wp-content/uploads/2022/09/architecture-diagram.png)
```
Comment (yzeng25) [inline blog/en/blog/2022/09/28/rust-loves-apisix.md:59]: Probably need to covert to Tencent CDN, cc @juzhiyuan

File: blog/en/blog/2022/09/28/rust-loves-apisix.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+### The code itself
+
+Let me first say that the available documentation is pretty sparse. For example, `proxy-wasm`'s is limited to the [methods' signature](https://github.com/proxy-wasm/spec/tree/master/abi-versions/vNEXT) (think JavaDocs). Rust SDK is sample-based. However, one can get some information from the [C++ SDK](https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/blob/master/docs/wasm_filter.md).
+
+>WASM module is running in a stack-based virtual machine and its memory is isolated from the host environment. All interactions between host and WASM module are through functions and callbacks wrapped by context object.
+>
+>At bootstrap time, a root context is created. The root context has the same lifetime as the VM/runtime instance and acts as a target for any interactions which happen at initial setup. It is also used for interactions that outlive a request.
+>
+>At request time, a context with incremental is created for each stream. Stream context has the same lifetime as the stream itself and acts as a target for interactions that are local to that stream.
```
Comment (yzeng25) [inline blog/en/blog/2022/09/28/rust-loves-apisix.md]: ditto
```suggestion
> WASM module is running in a stack-based virtual machine and its memory is isolated from the host environment. All interactions between host and WASM module are through functions and callbacks wrapped by context object.
>
> At bootstrap time, a root context is created. The root context has the same lifetime as the VM/runtime instance and acts as a target for any interactions which happen at initial setup. It is also used for interactions that outlive a request.
>
> At request time, a context with incremental is created for each stream. Stream context has the same lifetime as the stream itself and acts as a target for interactions that are local to that stream.
```

File: blog/en/blog/2022/09/28/rust-loves-apisix.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+The Rust code maps to the same abstractions.
+
+![Rust's 'structure diagram'](https://api7.ai/wp-content/uploads/2022/09/struct-diagram.png)
```
Comment (yzeng25) [inline blog/en/blog/2022/09/28/rust-loves-apisix.md:106]: ditto, Tencent CDN? cc @juzhiyuan 
```suggestion
![Rust's structure diagram](https://api7.ai/wp-content/uploads/2022/09/struct-diagram.png)
```

**Review by yzeng25 — COMMENTED** [review]
~You would need to modify `blog/en/config/picked-posts.json` file to generate the final online preview, see [previous pull request](https://github.com/apache/apisix-website/pull/1337/files) for detail.~

~We can further review when that is done.~

Never mind, `blog/en/config/picked-posts.json` is for particular posts only. This pull request is good for now.

File: blog/en/blog/2022/09/28/rust-loves-apisix.md (hunk 4 above)
Comment (nfrankel) [reply blog/en/blog/2022/09/28/rust-loves-apisix.md:106]: There's no such thing as a structure diagram in UML, hence the quotes. I can remove them, I don't understand why.

File: blog/en/blog/2022/09/28/rust-loves-apisix.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+## Let's code!
+
+Now that we have explained how everything fits together, it's time to code.
+
+### Preparing Rust for WebAssembly
+
+Before developing the first line of code, we need to give Rust <abbr title="WebAssembly">WASM</abbr> compilation capabilities.
+
+```bash
```
Comment (SylviaBABY) [inline blog/en/blog/2022/09/28/rust-loves-apisix.md]:
```suggestion
```shell
```

Comment (SylviaBABY) [reply blog/en/blog/2022/09/28/rust-loves-apisix.md]: need to change the `bash` code type to `shell` in the rest of the content. Others LGTM

File: blog/en/blog/2022/09/28/rust-loves-apisix.md (hunk 4 above)
Comment (yzeng25) [reply blog/en/blog/2022/09/28/rust-loves-apisix.md:106]: OK cool.

**Review by yzeng25 — APPROVED** [review]
LGTM. Just one last thing to do or not: convert images. cc @SylviaBABY @juzhiyuan

**Comment by nfrankel** [comment]
@yzeng25 It will be good if you do it because I've no clue what you mean 😅

**Comment by yzeng25** [comment]
> @yzeng25 It will be good if you do it because I've no clue what you mean 😅

I'd love to but currently I don't have access to it. I meant converting the images' link to something like `https://static.apiseven.com/2022/09/26/example.png`, which means storing it in another place, just like the other images in the blog.

**Review by SylviaBABY — APPROVED** [review]

File: blog/en/blog/2022/09/28/rust-loves-apisix.md (hunk 2 above)
Comment (juzhiyuan) [reply blog/en/blog/2022/09/28/rust-loves-apisix.md:59]: No worry :)

**Pull request merged**